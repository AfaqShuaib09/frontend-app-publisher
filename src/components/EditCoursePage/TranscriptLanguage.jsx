import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';

import RenderSelectField from '../RenderSelectField';
import RemoveButton from '../RemoveButton';

class TranscriptLanguage extends React.Component {
  constructor(props) {
    super(props);
    this.handleRemove = this.handleRemove.bind(this);
  }

  handleRemove(index) {
    this.props.fields.remove(index);
  }

  render() {
    const { fields, languageOptions, disabled } = this.props;

    return (
      <div className="transcript-languages mb-3">
        <ul className="list-group p-0 m-0">
          {fields.map((language, index) => (
            <li className="transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0" key={language}>
              <Field
                name={`${language}`}
                component={RenderSelectField}
                options={languageOptions}
                type="text"
                disabled={disabled}
              />
              <RemoveButton
                label="Remove language"
                onRemove={this.handleRemove}
                targetFieldNumber={index}
                disabled={disabled}
              />
            </li>
          ))}
        </ul>
        <button
          type="button"
          className="btn btn-outline-primary js-add-button"
          onClick={() => fields.push({})}
          disabled={disabled}
        >
          Add language
        </button>
      </div>
    );
  }
}

TranscriptLanguage.propTypes = {
  fields: PropTypes.shape({
    remove: PropTypes.func,
  }).isRequired,
  languageOptions: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })).isRequired,
  disabled: PropTypes.bool,
};

TranscriptLanguage.defaultProps = {
  disabled: false,
};

export default TranscriptLanguage;
