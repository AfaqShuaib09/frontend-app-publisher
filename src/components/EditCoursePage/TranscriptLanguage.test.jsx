import React from 'react';
import { shallow } from 'enzyme';

import TranscriptLanguage from './TranscriptLanguage';

const languageOptions = [{
  label: 'Arabic - United Arab Emirates',
  value: 'ar-ae',
}];

describe('Transcript Language', () => {
  it('renders correctly with no fields', () => {
    const component = shallow(<TranscriptLanguage
      fields={[]}
      languageOptions={languageOptions}
    />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when given fields', () => {
    const component = shallow(<TranscriptLanguage
      fields={[{}]}
      languageOptions={languageOptions}
    />);
    expect(component).toMatchSnapshot();
  });

  it('adds fields when the add button is pushed', () => {
    const fields = [{}];
    const component = shallow(<TranscriptLanguage
      fields={fields}
      languageOptions={languageOptions}
    />);
    expect(fields.length).toEqual(1);

    component.find('.js-add-button').simulate('click');
    expect(fields.length).toEqual(2);
  });
});
