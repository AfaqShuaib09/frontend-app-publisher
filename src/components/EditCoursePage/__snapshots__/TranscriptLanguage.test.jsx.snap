// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transcript Language renders correctly when given fields 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranscriptLanguage
    disabled={false}
    fields={
      Array [
        Object {},
      ]
    }
    languageOptions={
      Array [
        Object {
          "label": "Arabic - United Arab Emirates",
          "value": "ar-ae",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="list-group p-0 m-0"
        >
          <li
            className="transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0"
          >
            <Field
              component={[Function]}
              disabled={false}
              name="[object Object]"
              options={
                Array [
                  Object {
                    "label": "Arabic - United Arab Emirates",
                    "value": "ar-ae",
                  },
                ]
              }
              type="text"
            />
            <RemoveButton
              className=""
              disabled={false}
              label="Remove language"
              onRemove={[Function]}
              targetFieldNumber={0}
            />
          </li>
        </ul>,
        <button
          className="btn btn-outline-primary js-add-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add language
        </button>,
      ],
      "className": "transcript-languages mb-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0"
            >
              <Field
                component={[Function]}
                disabled={false}
                name="[object Object]"
                options={
                  Array [
                    Object {
                      "label": "Arabic - United Arab Emirates",
                      "value": "ar-ae",
                    },
                  ]
                }
                type="text"
              />
              <RemoveButton
                className=""
                disabled={false}
                label="Remove language"
                onRemove={[Function]}
                targetFieldNumber={0}
              />
            </li>,
          ],
          "className": "list-group p-0 m-0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "[object Object]",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Field
                  component={[Function]}
                  disabled={false}
                  name="[object Object]"
                  options={
                    Array [
                      Object {
                        "label": "Arabic - United Arab Emirates",
                        "value": "ar-ae",
                      },
                    ]
                  }
                  type="text"
                />,
                <RemoveButton
                  className=""
                  disabled={false}
                  label="Remove language"
                  onRemove={[Function]}
                  targetFieldNumber={0}
                />,
              ],
              "className": "transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "disabled": false,
                  "name": "[object Object]",
                  "options": Array [
                    Object {
                      "label": "Arabic - United Arab Emirates",
                      "value": "ar-ae",
                    },
                  ],
                  "type": "text",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "",
                  "disabled": false,
                  "label": "Remove language",
                  "onRemove": [Function],
                  "targetFieldNumber": 0,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add language",
          "className": "btn btn-outline-primary js-add-button",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Add language",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="list-group p-0 m-0"
          >
            <li
              className="transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0"
            >
              <Field
                component={[Function]}
                disabled={false}
                name="[object Object]"
                options={
                  Array [
                    Object {
                      "label": "Arabic - United Arab Emirates",
                      "value": "ar-ae",
                    },
                  ]
                }
                type="text"
              />
              <RemoveButton
                className=""
                disabled={false}
                label="Remove language"
                onRemove={[Function]}
                targetFieldNumber={0}
              />
            </li>
          </ul>,
          <button
            className="btn btn-outline-primary js-add-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add language
          </button>,
        ],
        "className": "transcript-languages mb-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0"
              >
                <Field
                  component={[Function]}
                  disabled={false}
                  name="[object Object]"
                  options={
                    Array [
                      Object {
                        "label": "Arabic - United Arab Emirates",
                        "value": "ar-ae",
                      },
                    ]
                  }
                  type="text"
                />
                <RemoveButton
                  className=""
                  disabled={false}
                  label="Remove language"
                  onRemove={[Function]}
                  targetFieldNumber={0}
                />
              </li>,
            ],
            "className": "list-group p-0 m-0",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "[object Object]",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Field
                    component={[Function]}
                    disabled={false}
                    name="[object Object]"
                    options={
                      Array [
                        Object {
                          "label": "Arabic - United Arab Emirates",
                          "value": "ar-ae",
                        },
                      ]
                    }
                    type="text"
                  />,
                  <RemoveButton
                    className=""
                    disabled={false}
                    label="Remove language"
                    onRemove={[Function]}
                    targetFieldNumber={0}
                  />,
                ],
                "className": "transcript-language list-group-item d-flex align-items-center border-0 p-0 m-0",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "component": [Function],
                    "disabled": false,
                    "name": "[object Object]",
                    "options": Array [
                      Object {
                        "label": "Arabic - United Arab Emirates",
                        "value": "ar-ae",
                      },
                    ],
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "",
                    "disabled": false,
                    "label": "Remove language",
                    "onRemove": [Function],
                    "targetFieldNumber": 0,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add language",
            "className": "btn btn-outline-primary js-add-button",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Add language",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Transcript Language renders correctly with no fields 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranscriptLanguage
    disabled={false}
    fields={Array []}
    languageOptions={
      Array [
        Object {
          "label": "Arabic - United Arab Emirates",
          "value": "ar-ae",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ul
          className="list-group p-0 m-0"
        />,
        <button
          className="btn btn-outline-primary js-add-button"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Add language
        </button>,
      ],
      "className": "transcript-languages mb-3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "list-group p-0 m-0",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add language",
          "className": "btn btn-outline-primary js-add-button",
          "disabled": false,
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Add language",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="list-group p-0 m-0"
          />,
          <button
            className="btn btn-outline-primary js-add-button"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Add language
          </button>,
        ],
        "className": "transcript-languages mb-3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "list-group p-0 m-0",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add language",
            "className": "btn btn-outline-primary js-add-button",
            "disabled": false,
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Add language",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
